@shared/api/helpers/getCollectibles.ts

```diff
@@ -0,0 +1,153 @@
+import { captureException } from "@sentry/browser";
+import {
+  CollectiblesResponse,
+  Collection,
+  CollectibleMetadata,
+} from "../types";
+import { NetworkDetails } from "@shared/constants/stellar";
+import { INDEXER_V2_URL } from "@shared/constants/mercury";
+
+/**
+ * Fetches metadata for a collectible from its token URI.
+ * The metadata follows the NFT metadata standard and includes name, description,
+ * image, external URL, and attributes.
+ *
+ * @param {string} tokenUri - The URI where the collectible metadata is hosted
+ * @returns {Promise<CollectibleMetadata | null>} The collectible metadata object, or null if the fetch fails or the response is invalid
+ */
+const fetchCollectibleMetadata = async (tokenUri: string) => {
+  let metadata: CollectibleMetadata | null = {
+    name: "",
+    description: "",
+    image: "",
+    externalUrl: "",
+    attributes: [],
+  };
+
+  try {
+    const response = await fetch(tokenUri);
+    if (!response.ok) {
+      return null;
+    }
+    const data = (await response.json()) as {
+      name?: string;
+      description?: string;
+      external_url?: string;
+      image?: string;
+      attributes?: {
+        traitType?: string;
+        value?: string | number;
+      }[];
+    };
```

Comment:
```suggestion
    const data = (await response.json()) as CollectibleMetadata;
```

Reply:
Ah, I actually can't cast this as `CollectibleMetadata` because `CollectibleMetadata` expects camelCasing while the API returns snake_case. I can create a `CollectibleMetadataReponse` interface like I have with the other API responses and use that here instead